radsat-sk/operation/subsystems/RAntenna.c

```diff
@@ -0,0 +1,293 @@
+				return error;
+			}
+
+			// Request Antenna Status
+			error = IsisAntS_getStatusData(ANTENNA_I2C_SLAVE_ADDR_PRIMARY,isisants_sideB,&RISISantsStatus);
+
+			// Error check for antenna status
+			if(error != 0) {
+
+				// TODO: record errors (if present) to System Manager
+				return error;
+			}
+
+			// Update Armed Section of struct
+			RAntennaStatus.AntennaArmed = RISISantsStatus.fields.armed;
+
+			// Must be armed before deployment can occur
+			if (RAntennaStatus.AntennaArmed) {
+
+				// Start automatic deployment
+				int error = IsisAntS_autoDeployment(ANTENNA_I2C_SLAVE_ADDR_PRIMARY,isisants_sideB,MAX_DEPLOYMENT_TIMEOUT);
+
+				// Check if autoDeployment failed
+				if(error != 0) {
+
+					// TODO: record errors (if present) to System Manager
+					return error;
+				}
+
+				//disarm B Side Antenna system
+				error = IsisAntS_setArmStatus(ANTENNA_I2C_SLAVE_ADDR_PRIMARY,isisants_sideB,isisants_disarm);
+
+				// Check if disarm failed
+				if(error != 0) {
+
+					// TODO: record errors (if present) to System Manager
+					return error;
+				}
+			}
+
+		}
+		// Increment deployment attempts for side B
+		antennaDeploymentAttempts += 1;
+	}
+
+
+	return 0;
+}
+
+/**
+ * Collect all telemetry from the ISISpace Antenna which includes temperatures, uptime, and antenna status
+ *
+ * @param side 0 for top side of antenna temperature, 1 for bottom side of the antenna temperature, will default to bottom side
+ * @return 0 for success, non-zero for failure. See hal/errors.h for details.
+ */
+int antennaTelemetry(antenna_telemetry_t* telemetry) {
+
+	int error;
+
+	// ensure the pointer is valid
+	if (telemetry == 0)
+		return E_INPUT_POINTER_NULL;
+
+	// create ISIS Telemetry struct
+	ISISantsTelemetry RISISantsTelemetry = { .fields = { 0 } };
+
+	// Execute Telemetry command for Side A
+	error = IsisAntS_getAlltelemetry(ANTENNA_I2C_SLAVE_ADDR_PRIMARY, isisants_sideA, &RISISantsTelemetry);
+
+	// Error check for Isis Antenna function
+	if(error != 0) {
+
+		// TODO: record errors (if present) to System Manager
+		return error;
+	}
+
+	// Assign side A Telemetry to struct
+	telemetry->sideA.deployStatus.DeployedAntennaOne = !RISISantsTelemetry.fields.ants_deployment.fields.ant1Undeployed;
+	telemetry->sideA.deployStatus.DeployedAntennaTwo = !RISISantsTelemetry.fields.ants_deployment.fields.ant2Undeployed;
+	telemetry->sideA.deployStatus.DeployedAntennaThree = !RISISantsTelemetry.fields.ants_deployment.fields.ant3Undeployed;
+	telemetry->sideA.deployStatus.DeployedAntennaFour = !RISISantsTelemetry.fields.ants_deployment.fields.ant4Undeployed;
+	telemetry->sideA.deployStatus.AntennaArmed = !RISISantsTelemetry.fields.ants_deployment.fields.armed;
+	telemetry->sideA.board_temp = 0.00322581 * RISISantsTelemetry.fields.ants_temperature;
+	telemetry->sideA.uptime = RISISantsTelemetry.fields.ants_uptime;
+
+	// Execute command for Side B
+	error = IsisAntS_getAlltelemetry(ANTENNA_I2C_SLAVE_ADDR_PRIMARY, isisants_sideB, &RISISantsTelemetry);
```

Comment:
Declare more explicit error names.

Reply:
I believe that having one error name is sufficient because in the case we do fail, the function will return no matter what which means that only one error variable is really necessary.